tests/testthat/test-exercise.R

```diff
@@ -557,7 +557,81 @@ test_that("error if env var directory does not exist", {
     ex <- mock_exercise(user_code = 'readLines("data/test.txt")')
     expect_error(
       evaluate_exercise(ex, new.env(), evaluate_global_setup = TRUE),
-      class = "learnr.missing_source_dir"
+      class = "learnr_missing_source_data_dir"
     )
   })
 })
+
+test_that("alternate data directory specified with `options()`", {
+  withr::with_tempdir({
```

Comment:
Thanks to the pattern you started in #544 https://github.com/rstudio/learnr/blob/bcd5dc3ba831bf44d255dc2643201610161dda32/R/exercise.R#L315-L318

I realized we can avoid putting the whole test in `with_tempdir()`

```suggestion
  withr::local_dir(withr::local_tempdir())
```

Reply:
This is totally worth the effort -- it makes it so much easier to step through the test interactively. withr is really cool 🤯

Reply:
Elegant!